### ISSUE TITLE
[question] How to make the vulkan validation layers package recipe export environment variable

### ISSUE THREAD

**Issue by Makogan** [conan-io/conan#9280]
At least on Linux, the vulkan sdk requires to run a setup script to export an environment variable, the relevant lines are:

```
VK_LAYER_PATH="$VULKAN_SDK/etc/vulkan/explicit_layer.d"
export VK_LAYER_PATH
```
This exposes a group of json files inside the sdk that are necessary for the validation layers to work properly. It is possible to export the same json file and allow proper funcitoning of the validation layers with conan by manually exporting the res folder, for example:

```
VK_LAYER_PATH=/home/makogan/.conan/data/vulkan-validationlayers/1.2.154.0/_/_/package/dd1b83909407d3b1c5d67ac14e74d4c40bceca38/res/vulkan/explicit_layer.d/
 export VK_LAYER_PATH
 ```
 
 Imitating the SDK's setup script. However this requires one to manually find the installation folder of the conan package, which is somewhat tedious. In the package_info method of the recipe, I can see that the environment variable is set:
 
 ```
     def package_info(self):

        self.cpp_info.libs = ["VkLayer_utils"]

        manifest_subfolder = "bin" if self.settings.os == "Windows" else os.path.join("res", "vulkan", "explicit_layer.d")

        vk_layer_path = os.path.join(self.package_folder, manifest_subfolder)

        self.output.info("Appending VK_LAYER_PATH environment variable: {}".format(vk_layer_path))

        self.env_info.VK_LAYER_PATH.append(vk_layer_path)
        ````

Upon calling conan install I see: 

```
vulkan-validationlayers/1.2.182: Appending VK_LAYER_PATH environment variable: /home/makogan/.conan/data/vulkan-validationlayers/1.2.182/_/_/package/3d88b9a6a78fe2fb3c381139b6734c20cf847d5b/res/vulkan/explicit_layer.d
```

But the path is not set upon installation (I print the variable and it is not set to the specified path). I am trying to set it manually in my own recipe but:
`self.deps_cpp_info["vulkan-validationlayers"].env_info.VK_LAYER_PATH`

Doesn't exist. I don;t quite understand how to export this environment variable. Moreover this value is reset upon terminal closure so I think I need to pass it onto meson to call it every time I compile. 

TL;DR how do I export the path to the validaiton layers from conan?

- [ x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

- [ x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @Makogan 

It is not possible to set it directly in your consumer, this is definitely not valid: ```self.deps_cpp_info["vulkan-validationlayers"].env_info.VK_LAYER_PATH``

I would need a bit more info about how you are consuming that package. Typically, a minimal reproducible case is very helpful (very simplified conanfile.py, one or several, in a folder, with the steps and exact commands to reproduce).

It might be that you are missing some ``virtualenv`` generator in the consumer side, that generates a script that can be activated to define the variables, but this is just a guess.

**Comment by Makogan** [comment]
This is my current conan.py:

```
from conans import ConanFile, tools, Meson
import os


class ConanFileToolsTest(ConanFile):
    generators = "pkg_config"
    requires = [
        "glfw/3.3.4",
        "shaderc/2019.0",
        "freetype/2.10.4",
        "harfbuzz/2.8.2",
        "vulkan-memory-allocator/2.3.0",
        "gtest/1.10.0",
        "benchmark/1.5.3",
        "tinygltf/2.5.0",
        "stb/20200203",
        "vulkan-headers/1.2.176",
        "vulkan-loader/1.2.182",
        "vulkan-validationlayers/1.2.182"
    ]
    settings = "os", "compiler", "build_type"

    def source(self):
        git = tools.Git(folder="libraries/imgui")
        git.clone("https://github.com/ocornut/imgui.git", "master")

        git = tools.Git(folder="libraries/Eigen")
        git.clone("https://gitlab.com/libeigen/eigen.git", "master")

        git = tools.Git(folder="libraries/incbin")
        git.clone("https://github.com/graphitemaster/incbin.git", "main")

    def build(self):
        env_script = f'export VK_LAYER_PATH={self.deps_env_info["vulkan-validationlayers"].VK_LAYER_PATH[0]}'
        with open('Scripts/env_script.sh', 'w+') as f:
            f.write(env_script)

        meson = Meson(self)
        meson.configure(build_folder="build")
        meson.build(args=['-j2'])
```
